experimental/zerotrie/src/byte_phf/mod.rs

```diff
@@ -0,0 +1,362 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[cfg(feature = "alloc")]
+mod builder;
+#[cfg(feature = "alloc")]
+mod cached_owned;
+
+#[cfg(feature = "alloc")]
+pub use builder::find;
+#[cfg(feature = "alloc")]
+pub use cached_owned::PerfectByteHashMapCacheOwned;
+
+const P_FAST_MAX: u8 = 11;
+const Q_FAST_MAX: u8 = 95;
+
+#[cfg(feature = "alloc")] // used in the builder code
+const P_REAL_MAX: u8 = 15;
+#[cfg(feature = "alloc")] // used in the builder code
+const Q_REAL_MAX: u8 = 127;
+
+/// Like slice::split_at but returns an Option instead of panicking
+#[inline]
+fn debug_split_at(slice: &[u8], mid: usize) -> Option<(&[u8], &[u8])> {
```

Comment:
issue: this is defined multiple times in your code

Reply:
Moved to helpers.rs